File: java/complete/myretro/src/main/java/com/apress/myretro/client/UserClient.java
package com.apress.myretro.client;

import com.apress.myretro.user.User;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

@Service
public class UserClient {
    private RestTemplate restTemplate;
    private String server;

    public UserClient(@Value("${user.server:http://localhost:8080}") String server) {
       this.restTemplate = new RestTemplate();
       this.server = server;
    }

    public User getUserByEmail(String email){
        ResponseEntity<User> responseEntity = this.restTemplate.getForEntity(this.server + "/users/" + email,User.class);
        return responseEntity.getBody();
    }
}

File: kotlin/complete/myretro/src/main/kotlin/com/apress/myretro/client/UserClient.kt
package com.apress.myretro.client

import com.apress.myretro.user.User
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import org.springframework.web.client.RestTemplate


@Service
class UserClient {

    private val restTemplate: RestTemplate = RestTemplate()
    private val server: String

    constructor(@Value("\${user.server:http://localhost:8080}") server: String) {
        this.server = server
    }

    fun getUserByEmail(email: String): User? {
        val responseEntity = restTemplate.getForEntity(
            "$server/users/$email",
            User::class.java
        )
        return responseEntity.body
    }
}